ads/google/imaVideo.js

```diff
@@ -221,241 +195,154 @@ let consentState;
 // Throttle for the showControls() function
 let showControlsThrottled = throttle(window, showControls, 1000);
 
+/**
+ * @param {!Node} parent
+ * @param {string} css
+ */
+function insertCss(parent, css) {
+  const style = parent.ownerDocument.createElement('style');
+  style./*OK*/ textContent = css;
+  parent.appendChild(style);
+}
+
+/**
+ * @param {string} state
+ * @param {boolean} active
+ */
+function toggleRootDataAttribute(state, active) {
+  const {'root': root} = elements;
+  const attributeName = `data-${state}`;
+  if (active) {
+    root.setAttribute(attributeName, '');
+  } else {
+    root.removeAttribute(attributeName);
+  }
+}
+
+/**
+ * @param {Document} elementOrDoc
+ * @return {!Object<string, !Element>}
+ */
+function renderElements(elementOrDoc) {
+  const html = htmlFor(elementOrDoc);
+
+  // Elements annotated with `ref="name"` are referenced as `elements['name']`.
+  // They also have their `ref` copied as a classname, so they can be selected
+  // from CSS using that exact value.
+  const root = html`
+    <div class="fill">
+      <div class="fill">
+        <video
+          ref="video"
+          playsinline
+          controlslist="nodownload nofullscreen noremoteplayback"
+        >
+          <!-- Video children are later propagated from the iframe's name. -->
+        </video>
+      </div>
+
+      <div ref="adContainer" class="fill" hidden>
+        <!-- This subtree may be modified by the IMA SDK. -->
+      </div>
+
+      <div ref="controls" hidden>
+        <button ref="playButton"></button>
```

Comment:
There are two structural differences in the new tree. The main one is how SVGs are handled.

SVGs have to be handled separately since they're created under a different HTML namespace. We now use `svgFor`, but we can also see how the namespace is declared in the now removed `createIcon()` function (`setAttributeNS`).

In order to insert the separately created nodes, we need to target their position. This is simpler with unique parents rather than position logic with `insertBefore`. They happen to be inserted in buttons, so we now create `<button>` wrappers for the SVGs. An existing `bigPlayDiv` was also converted to the `overlayButton` near the end of the tree.

I expect this to change the accessibility properties of the controls. My only assertion is that they can now be navigated by tab and clicked on enter. @caroqliu I would be very grateful if you take a look at a11y on the previous and the newer version. (I will share comparison links here later.)

Reply:
Happy to help! I'm taking this as directions to do nothing until links are available :)

Reply:
Right, thanks! I realized sharing a 3p frame build involved overriding some URLs on output binaries.

Here you go:

- [Before](https://amp-ima-video-34248.vercel.app/before)
- [After](https://amp-ima-video-34248.vercel.app/after)

Reply:
It also looks like the two versions have comparable Lighthouse scores for Accessibility. Before -
![image](https://user-images.githubusercontent.com/10456171/117719008-e0dbf400-b1aa-11eb-9aee-dab3aee86eb5.png)

After - 
![image](https://user-images.githubusercontent.com/10456171/117719023-e3d6e480-b1aa-11eb-84fe-88a3014ea991.png)